lib/taxon_description_updater.rb

```diff
@@ -0,0 +1,38 @@
+class TaxonDescriptionUpdater
+  def initialize(descriptions_to_remove)
+    @descriptions_to_remove = descriptions_to_remove
+  end
+
+  def call
+    @descriptions_to_remove.each(&method(:replace_description))
+  end
+
+private
+
+  EXCLUDE_ATTRIBUTES = %w[content_store user_facing_version publication_state lock_version updated_at state_history].freeze
+
+  def replace_description(description_to_remove)
+    content_items = publishing_api.get_content_items(
+      per_page: 5000,
+      q: description_to_remove,
+      search_in: ['description']
+    )
+    content_items['results'].select { |item| item['description'] == description_to_remove }.each do |taxon|
+      update_description(taxon)
+    end
+  end
+
+  def update_description(taxon)
+    content_id = taxon['content_id']
+    payload = taxon.except(*EXCLUDE_ATTRIBUTES).merge(
+      'description' => nil,
+      'update_type' => 'minor'
+    )
+    publishing_api.put_content(content_id, payload)
```

Comment:
I'm a bit concerned about the possibility that there may be some taxons which have a draft edition, which doesn't match the live edition in some way (e.g. title, phase, ...) and that running this script would publish those changes.

Would it be possible to put some checking in place, to make sure that this only automatically changes taxons where no draft previously existed? Or that it checks a few of the important user editable fields match between the draft and published editions prior to publishing?

Reply:
Now we only publish the description change if there is not an existing draft.